salt/states/boto_iam.py

```diff
@@ -0,0 +1,198 @@
+# -*- coding: utf-8 -*-
+'''
+Manage IAM roles.
+=================
+
+.. versionadded:: TBD
+
+This module uses ``boto``, which can be installed via package, or pip.
+
+This module accepts explicit IAM credentials but can also utilize
+IAM roles assigned to the instance through Instance Profiles. Dynamic
+credentials are then automatically obtained from AWS API and no further
+configuration is necessary. More information available `here
+<http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html>`_.
+
+It's also possible to specify ``key``, ``keyid`` and ``region`` via a profile, either
+passed in as a dict, or as a string to pull from pillars or minion config:
+
+.. code-block:: yaml
+
+    create-user:
+      boto_iam.user_present:
+        - name: myuser
+        - password: NewPassword$$1
+        - group: mygroup
+        - region: eu-west-1
+        - keyid: 'AKIAJHTMIQ2ASDFLASDF'
+        - key: 'fdkjsafkljsASSADFalkfjasdf'
+
+.. code-block:: yaml
+
+    creats-group:
+      boto_iam.group_present:
+        - name: mygroup
+        - users:
+          - myuser
+          - myuser1
+        - policy_name: test1
+        - policy: '{ "Version": "2012-10-17", "Statement": [ { "Effect": "Allow", "Action": "*", "Resource": "*" }]}'
+        - region: eu-west-1
+        - keyid: 'AKIAJHTMIQ2ASDFLASDF'
+        - key: 'safsdfsal;fdkjsafkljsASSADFalkfj'
+
+.. code-block:: yaml
+
+    change-policy:
+      boto_iam.account_policy:
+        - change_password: True
+'''
+
+import logging
+import json
+import salt.utils.odict as odict
+log = logging.getLogger(__name__)
+
+
+def __virtual__():
+    '''
+    Only load if boto is available.
+    '''
+    return 'boto_iam' if 'boto_iam.get_user' in __salt__ else False
+
+
+def user_present(name, password=None, path=None, group=None, region=None, key=None, keyid=None, profile=None):
```

Comment:
Can't assign policy to users, only groups? Probably not an issue for now, but would be nice to also have user policies :).